pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -560,6 +568,24 @@ public static Response getPinotQueryResponse(BrokerResponse brokerResponse)
         .build();
   }
 
+  // If a broker response has multiple exceptions, we will emit metrics for all of them.
+  // Thus, the sum total of all exceptions is >= total number of queries impacted.
+  // Additionally, some parts of code might already be emitting metrics for individual error codes.
+  // But that list isn't accurate with a many-to-many relationship (or no metrics) between error codes and metrics.
+  // This method ensures we emit metrics for all queries that have exceptions with a one-to-one mapping.
+  private void emitBrokerResponseMetrics(BrokerResponse brokerResponse) {
```

Comment:
This method should be moved to a common place so that it can be reused in BrokerGrpcServer and other places. 

what would be the right class for it ? BrokerResponse or BrokerMetrics or something else ?

Reply:
ideally would want to use the same code in server too so should I rather create a new class in org.apache.pinot.common.metrics for just this code ?

Reply:
Is metrics from servers useful ? The broker has the inteliigence to get all the exceptions. If a query is cancelled, then every server will report a cancellation.